packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -426,4 +425,32 @@ class AndroidLicenseValidator extends DoctorValidator {
     }
     return _processManager.canRun(sdkManagerPath);
   }
+
+  String messageForSdkManagerError(
+    Java java,
+    AndroidSdk androidSdk,
+    List<String> androidSdkStderr,
+    int exitCode,
+    Platform platform,
+  ) {
+    final String sdkManagerPath = androidSdk.sdkManagerPath ?? '';
+
+    final bool failedDueToJdkIncompatibility = androidSdkStderr.join().contains(
+      RegExp(r'.*java\.lang\.UnsupportedClassVersionError.*SdkManagerCli '
```

Comment:
nit, the `.*` at the start and end are redundant.

Reply:
Addressed in [57add0c](https://github.com/flutter/flutter/pull/138762/commits/57add0c0351c5291653afcffee5630e5055224ff)